### PR TITLE
DDP-5571 activity instance summary API

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

See DDP-5571. This PR implements the new API endpoint for fetching single activity instance summary ([see docs here](https://pepper-dev.datadonationplatform.org/docs#operation/getActivitySummary)).

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

- [x] :relaxed: All good, business as usual!

## How do we demo these changes?

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.

## Testing

- [x] I have written automated positive tests
- [x] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!

### CODE REVIEW

**Review by MocanaAtBroad — DISMISSED** [review]

**Comment by yufengwng** [comment]
@MocanaAtBroad I had to rebase this branch after the other PR was merged. Can I get another thumbs-up?

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceSummaryDto.java [hunk 1]
```diff
@@ -108,4 +111,20 @@ public boolean isHidden() {
...
+    public int getInstanceNumber() {
+        return instanceNumber;
+    }
+
+    public void setInstanceNumber(int instanceNumber) {
+        this.instanceNumber = instanceNumber;
+    }
+
+    public String getPreviousInstanceGuid() {
+        return previousInstanceGuid;
```
Comment (MocanaAtBroad) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceSummaryDto.java:124]: What is the purpose of `previousInstanceGuid`?

Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceSummaryDto.java:124]: I believe this was added for some features in A-T study.

Comment (MocanaAtBroad) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceSummaryDto.java:124]: 3478d5e76ea255330a6d2ba27814cf19

**Review by MocanaAtBroad — APPROVED** [review]

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceSummaryDto.java (hunk 1 above)
Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/db/dto/ActivityInstanceSummaryDto.java:124]: Just to be clear, this property already existed before this PR. The changes here is populating this property so we don't break existing functionality.

**Pull request merged**